processing/src/main/java/org/apache/druid/query/filter/Filter.java

```diff
@@ -37,6 +38,25 @@
 @SubclassesMustOverrideEqualsAndHashCode
 public interface Filter
 {
+  default String getFilterString()
+  {
+    return toString();
+  }
+
+  /**
+   * Returns an ordered set of all child filters for this filter with no duplicates. The default return value is an
+   * empty set.
+   * <p>
+   * Only {@link org.apache.druid.segment.filter.AndFilter} and {@link org.apache.druid.segment.filter.OrFilter} have
+   * child filters. Other filters would return empty (including {@link org.apache.druid.segment.filter.NotFilter}).
+   * <p>
+   * The ordering of child filters is important in some cases, e.x.short-curcuiting.
+   */
+  default ImmutableSet<Filter> getFilters()
```

Comment:
I still can't stop thinking that this method is strange here... what do you think about leaving this on `BooleanFilter` and just using `filter instance BooleanFilter` in `FilterBundle.Builder`?

We also should keep this as a `LinkedHashSet` since if the context flag is set to be off then the order specified is important because it was likely manually set to override the filter re-arranging.

Reply:
Ok reverted this change